src/Microsoft.DotNet.Build.Tasks/VersionTools/BaseDependenciesTask.cs

```diff
@@ -0,0 +1,163 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using Microsoft.DotNet.VersionTools;
+using Microsoft.DotNet.VersionTools.Dependencies;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.DotNet.Build.Tasks.VersionTools
+{
+    public abstract class BaseDependenciesTask : Task
+    {
+        [Required]
+        public ITaskItem[] DependencyBuildInfo { get; set; }
+
+        public ITaskItem[] ProjectJsonFiles { get; set; }
+
+        public ITaskItem[] XmlUpdateStep { get; set; }
+
+        public string BuildInfoCacheDir { get; set; }
+
+        public override bool Execute()
+        {
+            MsBuildTraceListener[] listeners = Trace.Listeners.AddMsBuildTraceListeners(Log);
+            try
+            {
+                TraceListenedExecute();
+            }
+            finally
+            {
+                Trace.Listeners.RemoveMsBuildTraceListeners(listeners);
+            }
+            return !Log.HasLoggedErrors;
+        }
+
+        protected abstract void TraceListenedExecute();
+
+        protected Regex CreateXmlUpdateRegex(string elementName, string contentGroupName) =>
+            new Regex($@"<{elementName}>(?<{contentGroupName}>.*)</{elementName}>");
+
+        protected IEnumerable<IDependencyUpdater> CreateUpdaters(IEnumerable<BuildInfo> buildInfos)
+        {
+            if (ProjectJsonFiles != null && ProjectJsonFiles.Any())
+            {
+                yield return new ProjectJsonUpdater(ProjectJsonFiles.Select(item => item.ItemSpec))
+                {
+                    // If SkipStableVersions is specified (true or non-true) on a build info item,
+                    // then make the updater follow that behavior for versions in that build info.
+                    BuildInfoSkipStableVersions = DependencyBuildInfo
+                        .Select(item => new
+                        {
+                            Info = buildInfos.Single(info => info.Name == item.ItemSpec),
+                            Skip = item.GetMetadata("SkipStableVersions")
```

Comment:
Yes I like your change to use UpdateStableVersions.

As for the convention I do think it is a good principle to always default values to the default/empty value if at all possible.